hnn_core/network.py

```diff
@@ -440,21 +464,29 @@ def add_poisson_drive(self, name, *, tstart=0, tstop=None, rate_constant,
                                                          weights_nmda)[0]
             _check_poisson_rates(rate_constant, target_populations,
                                  self.cell_types.keys())
-
         drive = _NetworkDrive()
         drive['type'] = 'poisson'
-        drive['cell_specific'] = True
+        if n_drive_cells == 'n_cells':
+            n_drive_cells = self.n_cells
+            drive['cell_specific'] = True
+        else:
+            _validate_type(n_drive_cells, types=int)
+            _check_option('n_drive_cells', n_drive_cells, allowed_values=[1])
+            drive['cell_specific'] = False
```

Comment:
I feel `cell_specific` (or should it be called `connectivity`?) should be exposed rather than this kind of implicit assumption. What if the user wants `n_drive_cells  = 'n_cells'` with all-to-all connectivity?

Reply:
This would be a rare use-case because that would blast the network with a ton of synchronous drives. However, if the user wants `n_drive_cells = 'n_cells'` with all-to-all connectivity, they should enter an integer amount such as `n_drive_cells = net.n_cells`.